### DISCUSSION TITLE
Hello, can someone explain what does "C and JavaScript backends" mean in the Key features of V?

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#20746] [Questions and Answers] [answered]
Hi, I'm new to programming. So, I don't quite understand what it is mean. Can someone please explain it to me? Thank you (^⁠‿⁠^⁠) 🙏 Have a nice day.

**Comment by JalonSolov** [comment, marked as answer]
By default, the V compiler outputs C code, then calls the C compiler for you to create the final binary, shared library, or whatever you told it to create.  It then deletes the C code to clean up disk space.

You can also tell V to output the C code then stop.  This is useful if you want to examine the C code itself, port to systems V doesn't currently support, etc.

V can also output JavaScript instead of C, which can be used with node, run in the browser, etc.

Last, there are other backends, such as the `native` backend (V compiles directly to a native executable, with no other compilers involved), `WASM` (Web Assembly), a `Go` backend, and someone was working on a FreePascal backend.  None of the backends are complete at the moment, other than the C backend.

The goal, once the `native` backend is complete, is for it to be the default, with the C backend as an option.  This way you would not need to have a C compiler installed at all, unless you want it.

**Comment by Wajinn** [comment]
It should be noted that V's JavaScript and WASM backends are significantly usable, though not yet complete. Depending on what a person is doing, they can compile/create working programs in those target languages. 

With Go, seems to be a bit of a different situation, where people might want to look more at Go2V (along with C2V).
